inc/genotype.hpp

```diff
@@ -65,6 +66,9 @@ class Genotype{
         std::list<Node> node_genes;
         std::list<Connection> connection_genes;
 
+        // adjacency list strcture of the network
+        std::map<int, std::list<int>> graph;
```

Comment:
1. Node ID type is uint64_t (check Node struct); Please be consistent and use uint64_t instead of int.
2. Consider using std::set<uint64_t> as the value type. Rationale: 1) Order does not matter 2) Obtaining all parents of a child node only takes O(NlogN) - N = # of nodes, whereas searching linearly through the edge list takes O(E) - E = # of edges, or in the worst case, O(N^2).

Reply:
Also note that we are constructing a weighted, directed graph. Thus we also need to record the weight of each edge in the adj list. So the type of graph should be:
`std::map<uint64_t, std::set<std::pair<uint64_t, long double>>>`

I would use a `using declaration`:
`using Graph = std::map<uint64_t, std::set<std::pair<uint64_t, long double>>>;` and
`Graph graph;`